microsoft/testsuites/xdp/xdptools.py

```diff
@@ -0,0 +1,134 @@
+        return str(self._command)
+
+    @property
+    def can_install(self) -> bool:
+        ethtool = self.node.tools[Ethtool]
+        statistics = ethtool.get_device_statistics(self.node.nics.default_nic)
+
+        # check if xdp supported on nic
+        if not any("xdp_drop" in x for x in statistics):
+            raise UnsupportedDistroException(
+                self.node.os,
+                "Cannot find xdp_drop in ethotool statistics. "
+                "It means it doesn't support XDP.",
+            )
+
+        return True
+
+    def run_full_test(self) -> None:
+        """
+        run full test of xdp tools repo
+        """
+        result = self.node.execute("make test", sudo=True, cwd=self._code_path)
+
+        abnormal_results: Dict[str, str] = {}
+        for item in find_groups_in_lines(
+            result.stdout, pattern=self._xdp_test_result_pattern
+        ):
+            if item["result"] not in ["PASS", "SKIPPED"]:
+                abnormal_results[item["name"]] = item["result"]
+        if abnormal_results:
+            raise LisaException(f"found failed tests: {abnormal_results}")
+        result.assert_exit_code(
+            0, "unknown error on xdp tests, please check log for more details."
+        )
+
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        super()._initialize(*args, **kwargs)
+        self._command: PurePath = PurePath(self._default_command)
+        self._gro_lro_settings: Dict[str, DeviceGroLroSettings] = {}
+
+    def _install(self) -> bool:
+        # install dependencies
+
+        config_envs: Dict[str, str] = {}
+        if isinstance(self.node.os, Debian):
+            if self.node.os.information.version < "18.4.0":
+                raise UnsupportedDistroException(self.node.os)
+            elif self.node.os.information.version == "18.4.0":
+                self.node.os.add_repository("ppa:ubuntu-toolchain-r/test")
+            else:
+                toolchain = f"llvm-toolchain-{self.node.os.information.codename}"
+                self.node.os.add_repository(
+                    repo=(
+                        f"deb http://apt.llvm.org/{self.node.os.information.codename}/ "
+                        f"{toolchain} main"
+                    ),
+                    key_location="https://apt.llvm.org/llvm-snapshot.gpg.key",
+                )
+
+            self.node.os.install_packages(
+                "clang-10 llvm libelf-dev libpcap-dev gcc-multilib build-essential "
+                "pkg-config m4 tshark"
+            )
+
+            config_envs.update({"CLANG": "clang-10", "LLC": "llc-10"})
+
+        elif isinstance(self.node.os, Fedora):
+            self.node.os.install_packages(
+                "llvm-toolset elfutils-devel m4 wireshark perf make gcc tc "
+                # pcaplib
+                "http://rpmfind.net/linux/centos/8.5.2111/PowerTools/"
+                "x86_64/os/Packages/libpcap-devel-1.9.1-5.el8.x86_64.rpm"
+            )
```

Comment:
seems it is only for centos 8.5 distro? if lower distro version used, will it fail?

Reply:
The tool is not 1:1 mapping, it's hard to meet all OS versions. I tested it with Redhat 8.4, it works. The Redhat 8.3 doesn't support xdp. so it should be ok.